src/Bridges/ApplicationLatte/TemplateFactory.php

```diff
@@ -31,14 +31,20 @@ class TemplateFactory implements UI\ITemplateFactory
 	/** @var Nette\Caching\IStorage */
 	private $cacheStorage;
 
+	/** @var string */
+	private $templateClassName;
+
 
 	public function __construct(ILatteFactory $latteFactory, Nette\Http\IRequest $httpRequest = NULL,
-		Nette\Security\User $user = NULL, Nette\Caching\IStorage $cacheStorage = NULL)
+		Nette\Security\User $user = NULL, Nette\Caching\IStorage $cacheStorage = NULL, $templateClassName = Template::class)
 	{
 		$this->latteFactory = $latteFactory;
 		$this->httpRequest = $httpRequest;
 		$this->user = $user;
 		$this->cacheStorage = $cacheStorage;
+
+		\Nette\Utils\Validators::assert($templateClassName, 'string');
```

Comment:
I have to do both checks because without checking string it's possible to call `new` on integer for example. From my point of view validation is needed. Can you help me how to prevent calling `$template = new $this->templateClassName($latte);` (line 57) with wrong `$this->templateClassName` type?